File: backend/app/schemas/conversation.py
"""Schemas for conversation history (US-16, FR-E2, FR-E6)."""
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class ConversationMessageOut(BaseModel):
    """A message in the conversation history."""

    id: int
    role: str  # "user" or "assistant"
    content: str
    created_at: datetime

    class Config:
        from_attributes = True


class ConversationSessionOut(BaseModel):
    """A conversation session with metadata and recent messages."""

    id: int
    student_id: int
    challenge_id: int | None
    theme_name: str | None
    message_count: int
    last_message_at: datetime
    created_at: datetime
    messages: list[ConversationMessageOut]

    class Config:
        from_attributes = True


class ConversationSessionCreate(BaseModel):
    """Request to create or resume a conversation session."""

    challenge_id: int | None = None

File: frontend/src/types/conversation.ts
/** Conversation history types for wellness guide (US-16, FR-E2, FR-E6) */

export interface ConversationMessage {
  id: number;
  role: "user" | "assistant";
  content: string;
  created_at: string;
}

export interface ConversationSession {
  id: number;
  student_id: number;
  challenge_id: number | null;
  theme_name: string | null;
  message_count: number;
  last_message_at: string;
  created_at: string;
  messages: ConversationMessage[];
}

export interface ConversationSessionCreate {
  challenge_id?: number | null;
}
